instrumentation/micrometer/micrometer-1.5/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/LongTaskTimerTest.java

```diff
@@ -0,0 +1,217 @@
+        metrics ->
+            metrics.anySatisfy(
+                metric ->
+                    assertThat(metric)
+                        .hasDescription("This is a test long task timer")
+                        .hasUnit("tasks")
+                        .hasLongSum()
+                        .isNotMonotonic()
+                        .points()
+                        .satisfiesExactly(
+                            point ->
+                                assertThat(point)
+                                    .hasValue(1)
+                                    .attributes()
+                                    .containsOnly(attributeEntry("tag", "value")))));
+
+    // when
+    TimeUnit.MILLISECONDS.sleep(100);
+    sample.stop();
+
+    // then
+    testing.waitAndAssertMetrics(
+        INSTRUMENTATION_NAME,
+        "testLongTaskTimer",
+        metrics ->
+            metrics.anySatisfy(
+                metric ->
+                    assertThat(metric)
+                        .hasDescription("This is a test long task timer")
+                        .hasUnit("ms")
+                        .hasDoubleHistogram()
+                        .points()
+                        .satisfiesExactly(
+                            point ->
+                                assertThat(point)
+                                    /* .hasSumGreaterThan(100) */
+                                    .hasCount(1)
+                                    .attributes()
+                                    .containsOnly(attributeEntry("tag", "value")))));
+    testing.waitAndAssertMetrics(
+        INSTRUMENTATION_NAME,
+        "testLongTaskTimer.active",
+        metrics ->
+            metrics.anySatisfy(
+                metric ->
+                    assertThat(metric)
+                        .hasLongSum()
+                        .points()
+                        .satisfiesExactly(
+                            point ->
+                                assertThat(point)
+                                    .hasValue(0)
+                                    .attributes()
+                                    .containsOnly(attributeEntry("tag", "value")))));
+    testing.clearData();
+
+    // when
+    Metrics.globalRegistry.remove(timer);
+    timer.start();
+
+    // then
+    testing.waitAndAssertMetrics(
+        INSTRUMENTATION_NAME,
+        "testLongTaskTimer.active",
+        metrics ->
+            metrics.anySatisfy(
+                metric ->
+                    assertThat(metric)
+                        .hasLongSum()
+                        .points()
+                        .satisfiesExactly(
+                            point ->
+                                assertThat(point)
+                                    .hasValue(0)
+                                    .attributes()
+                                    .containsOnly(attributeEntry("tag", "value")))));
+
+    // when
+    TimeUnit.MILLISECONDS.sleep(100);
+    sample.stop();
```

Comment:
did you mean to assign sample to `timer.start()` above?

Reply:
Nice catch! I've noticed that `Sample#stop()` was a bit broken thanks to that - fixed that and added one more test. Thanks!